Description: Use correct gdb logging command for newer gdb versions
Author: Fredrick Brennan <copypaste@kittens.ph>
Last-Update: 2025-04-20

--- a/dis-gdb.fs.orig
+++ b/dis-gdb.fs
@@ -44,11 +44,11 @@ defer gdb-addr-sep-char ( -- c )
 
 : disasm-gdb { addr u -- }
     base @ >r hex
-    s\" type mktemp >/dev/null && type gdb >/dev/null && file=`mktemp -t gforthdis.XXXXXXXXXX` && file2=`mktemp -t gforthdis.XXXXXXXXXX` && echo \"set verbose off\nset logging file $file\nset logging on\ndisas " save-mem ( addr u addr1 u1 )
+    s\" type mktemp >/dev/null && type gdb >/dev/null && file=`mktemp -t gforthdis.XXXXXXXXXX` && file2=`mktemp -t gforthdis.XXXXXXXXXX` && echo \"set verbose off\nset logging file $file\nset logging enabled on\ndisas " save-mem ( addr u addr1 u1 )
     addr 0 <<# gdb-addr-sep-char hold # #s 'x hold # #> append-extend-string #>>
     addr u + 0 <<# # #s 'x hold # #> append-extend-string #>>
     r> base ! cr
-    s\" \nset logging off\nquit\n\" >$file2 && gdb -nx -q -p `ps -p $$ -o ppid=` -x $file2 2>/dev/null >/dev/null && rm $file2 && grep -v \"of assembler\" $file && rm $file" append-extend-string
+    s\" \nset logging enabled off\nquit\n\" >$file2 && gdb -nx -q -p `ps -p $$ -o ppid=` -x $file2 2>/dev/null >/dev/null && rm $file2 && grep -v \"of assembler\" $file && rm $file" append-extend-string
     2dup (system) 2swap drop free throw throw if
 	addr u dump
     endif ;
